### DISCUSSION TITLE
Anyway to know once nivo chart has finished rendering?

### DISCUSSION THREAD

**Discussion by Cpt-Falcon** [plouc/nivo#2036] [General]
I've got a situation where I'm trying to take a screenshot of a nivo chart. There is an open source react project that enables the react client to take a screenshot of one of its own components. Useful for generating reports. The problem is that nivo chart isn't rendered right away. I need to wait like 100 ms then take a screenshot in order to have a picture. Is there an event or anything I could use that would tell me when rendering is complete so I could take the screenshot?

**Comment by frederickf** [comment]
I solved this by following the suggestion provided in https://github.com/plouc/nivo/issues/527. The issue also links to this example which show more detail https://codesandbox.io/s/won5y6k70w?file=/src/BarDemo.jsx

Basically, add a layer that executes when all the other layers are finished

```
import { BarDefaultProps, Bar } from '@nivo/bar';
const layers = [
  ...BarDefaultProps.layers,
  () => console.log('chart has been rendered!')
];
```

**Comment by tkonopka** [comment]
Hi @Cpt-Falcon. Does the layer-based approach work for you, and if not, could you create a small example that shows the problem, please?

I tried to save some charts as images. Capturing images from a useEffect block seemed to produce reasonable results. I have seen some hiccups with incomplete bar charts when 'animate' is on, and in those cases it can help to trigger a re-rendering before saving the image. If you have an example to debug, that would be useful.
